src/libraries/Microsoft.Extensions.Hosting/src/Microsoft.Extensions.Hosting.csproj

```diff
@@ -1,16 +1,28 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0;netstandard2.1;net461</TargetFrameworks>
+    <TargetFrameworks>$(NetCoreAppCurrent);netstandard2.0;netstandard2.1;net461</TargetFrameworks>
     <EnableDefaultItems>true</EnableDefaultItems>
     <PackageDescription>Hosting and startup infrastructures for applications.</PackageDescription>
+    <!-- Use targeting pack references instead of granular ones in the project file. -->
+    <DisableImplicitAssemblyReferences>false</DisableImplicitAssemblyReferences>
   </PropertyGroup>
 
-  <ItemGroup>
+  <ItemGroup Condition="!$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net6.0'))">
+    <Compile Remove="Internal\ConsoleLifetime.netcoreapp.cs" />
+  </ItemGroup>
+
+  <ItemGroup Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net6.0'))">
+    <Compile Remove="Internal\ConsoleLifetime.notnetcoreapp.cs" />
```

Comment:
This naming 🤣

Reply:
Suggestion for a better name?

Reply:
`.netstandard.cs` ?

Reply:
This library also builds for `net461` and that TFM will compile this file. Is `netstandard.cs` used elsewhere when targeting net461? I can use that name if it makes sense to people.

Aside: I plan on proposing that in .NET 7 we change `Microsoft.Extensions.Hosting` to only target .NET 6+ going forward. `Microsoft.Extensions.Hosting` is an app model that should only be referenced by executables, libraries shouldn't reference it. So just like winforms/wpf/aspnetcore/maui/etc, we should only need to target .NET 6+ in Hosting.